src/advanced-bloom/AdvancedBloomFilter.ts

```diff
@@ -73,11 +99,35 @@ export class AdvancedBloomFilter extends Filter
     private _extractFilter: ExtractBrightnessFilter;
     private _blurFilter: KawaseBlurFilter;
 
-    constructor(options?: AdvancedBloomFilterOptions)
+    constructor(options?: AdvancedBloomFilterOptions);
+    /**
+     * @deprecated since 8.0.0
+     *
+     * @param {object|number} [options] - The optional parameters of advanced bloom filter.
+     *                        When options is a number , it will be `options.threshold`.
+     * @param {number} [options.threshold=0.5] - Defines how bright a color needs to be to affect bloom.
+     * @param {number} [options.bloomScale=1.0] - To adjust the strength of the bloom. Higher values is
+     *        more intense brightness.
+     * @param {number} [options.brightness=1.0] - The brightness, lower value is more subtle brightness,
+     *        higher value is blown-out.
+     * @param {number} [options.blur=8] - Sets the strength of the Blur properties simultaneously
+     * @param {number} [options.quality=4] - The quality of the Blur filter.
+     * @param {number[]} [options.kernels=null] - The kernels of the Blur filter.
+     * @param {number|number[]|PIXI.PointData} [options.pixelSize=1] - the pixelSize of the Blur filter.
+     * @param {number} [options.resolution=PIXI.settings.FILTER_RESOLUTION] - The resolution of the Blur filter.
+     */
+    constructor(options?: Partial<DeprecatedAdvancedBloomFilterOptions>);
+    constructor(options?: AdvancedBloomFilterOptions | Partial<DeprecatedAdvancedBloomFilterOptions>)
```

Comment:
Why is there a mix of using `Partial<>` here and not? These interfaces are more useful if we keep the properties optional instead of using Partial so that users can cast:

```ts
const options = { pixelSize: 10 } as DeprecatedAdvancedBloomFilterOptions;
```